source/binder.cpp

```diff
@@ -13,6 +13,18 @@
 #include <binder.hpp>
 
 // Declares clang::SyntaxOnlyAction.
+#if  (LLVM_VERSION_MAJOR < 4)
```

Comment:
Please say a few words what range of version of LLVM do you propose Binder should support? We moved to use LLVM-4 almost immediately so i am not sure if supporting versions <4 makes much sense. Also, looks like most of if/def spaghetti code below is caused by supporting version below 4. So i think it would make more sense to just support only LLVM-4 and above.

Reply:
The absolutely lowest range I've tested is 3.4 just because it is the standard version in RHEL7 and I've started the porting from that version.  The versions between 4.0 and 7.0 don't need ifdefs. The versions above 7 and below 4.0 need some macros. The exact range should be up to you. I can only suggest to bind the versions of binder to the versions 
of "default" LLVM and define in docs what i "supported" and what "provided as is". But that is just a suggestion.